### ISSUE TITLE
Textalignment and Autoshrinking

### ISSUE THREAD

**Issue by MFJones21** [kostub/MathEditor#12]
So there are two things that I want to achieve with the MTEditableMathLabel.

1. Change the Textalignment to left

2. Make the Label expand / shrink with the latex inside. I have this code from the example and converted it to Swift 3

`func textModified(_ label: MTEditableMathLabel!) {
        let minHeight: CGFloat = 64
        // Increase the height of the label as the height increases.
        let mathSize: CGSize = label.mathDisplaySize()
        
        if (mathSize.height > self.editLabel.frame.height - 10) {
            label.layoutIfNeeded()
            // animate

            self.editLabel.frame = CGRect(x: self.editLabel.frame.origin.x, y: self.editLabel.frame.origin.y, width: self.editLabel.frame.width, height: mathSize.height + 10)
            label.layoutIfNeeded()
        }
        else if (mathSize.height < self.editLabel.frame.height - 20) {
            
            let newHeight: CGFloat = max(mathSize.height + 10, minHeight);
            if (newHeight < self.editLabel.frame.height) {
                label.layoutIfNeeded()
                self.editLabel.frame = CGRect(x: self.editLabel.frame.origin.x, y: self.editLabel.frame.origin.y, width: self.editLabel.frame.width, height: newHeight)
                label.layoutIfNeeded()
            }
        }
        
        // Shrink the font as the label gets longer.
        if (mathSize.width > label.frame.size.width - 10) {
            label.fontSize = label.fontSize * 0.9;
            
        } else if (mathSize.width < label.frame.size.width - 40) {
            let fontSize: CGFloat = min(label.fontSize * 1.1, 30);
            if (fontSize > label.fontSize) {
                label.fontSize = fontSize;
            }
        }
    }`

Do you have code do what this code is doing with the height with the width ?

**Comment by kostub** [comment]
See the code in:

https://github.com/kostub/MathEditor/blob/master/MathEditorExample/MTViewController.m

Maybe it can be moved to the `MTEditableMathLabel` and made an option?

**Comment by MFJones21** [comment]
So I got the auto shrinking the way I wanted it with some adjustemnts to the code from the example. Thanks =) So great ! The two issues I was talking about are now resolved. 

But I have a few things left to achieve with your awesome Project.

I have my shrieked label in a Scrollview and I want to programatically scroll the scrollview with the index in the Label. So is there a way to get the Graphical location of the index indicator in the label ?

I need to display both Integrals and Matrices. I asked this before and you told me that I will have to implement a Custom keyboard for that. I did but this doesn't work:

`- (void)keyPressed:(id)sender
{
    [self playClickForCustomKeyTap];
    [self.textView insertText:@"\\begin{Bmatrix} a_1 & a_2 & a_3 & a_4 \\ b_1 & b_2 & b_3 & b_4 \\ c_1 & c_2 & c_3 & c_4 \\ d_1 & d_2 & d_3 & d_4 \\end{Bmatrix}];
}`

and this doesn't work either

`- (void)keyPressed:(id)sender
{
    [self playClickForCustomKeyTap];
    [self.textView insertText:@"\\int ^{a}_{b}"];
}`

All that is getting displayed is nothing. Whats wrong ?
